assets/js/main.js

```diff
@@ -20,26 +30,43 @@ $(function() { /* Wait for jQuery */
     }
   });
 
+  /* Reset buttons */
+  $(".copy-but").prop("disabled", true);
+  $(".export .copy-but").text("Copy !");
+  $(".dl-but button").prop("disabled", true);
+  $(".dl-but select").prop("disabled", true);
+
   /* Compile button click */
   $(".process-but button").click(function() {
-    var duckyScript = $(".input > textarea").val();
-    var selectedModule = $(".process-but select").find(":selected").text();
+    let duckyScript = $(".input > #input").val();
+    let selectedModule = mods[$(".process-but select").find(":selected").val()];
 
     /* Load Duckuino & Compile */
-    Duck.loadModule(selectedModule);
-    var compilerOut = Duck.compileCode(duckyScript);
+    let compilerOut = Duck.compileCode(duckyScript, selectedModule.module);
 
     /* Check for error */
     if(compilerOut.returnCode === 0) {
       /* Set textarea text */
       $(".export > textarea").val(compilerOut.compiledCode);
 
       /* Enable buttons */
-      $(".dl-but button").prop("disabled", false);
-      $(".dl-but select").prop("disabled", false);
       $(".copy-but").prop("disabled", false);
       $(".export .copy-but").text("Copy !");
 
+      /* Reset & (Re)Populate locales */
+      $(".dl-but select").empty();
+      $(".dl-but button").prop("disabled", true);
+      $(".dl-but select").prop("disabled", true);
+      for (let y in selectedModule.meta.locales) {
+        let l = selectedModule.meta.locales[y];
+        if (y == "_meta")
+          continue;
+        $(".dl-but select").append("<option value=\"" + y + "\">" + l.name + "</option>");
```

Comment:
Suspicious this is XSSable

Reply:
Since the client has no write permission, how could this lead to XSS ?

Reply:
Ah derp, misread this. The html building via string made me nervous :P

Reply:
Np :D It's always read-only so no worries about crappy `eval()` and XSS :3